src/array/kernel.cc

```diff
@@ -83,6 +130,50 @@ void SDDMM(const std::string& op,
   });
 }
 
+
+/*! \brief Generalized Sampled Dense-Dense Matrix Multiplication. */
+void SDDMMHetero(const std::string& op,
+           HeteroGraphPtr graph,
+           std::vector<NDArray> lhs,
+           std::vector<NDArray> rhs,
+           std::vector<NDArray> out,
+           int lhs_target,
+           int rhs_target) {
+  // TODO(Israt): change it to COO_CODE
+  SparseFormat format = graph->SelectFormat(0, CSR_CODE);
```

Comment:
this is to change the format of the graph for relation type 0?

Reply:
Yes. We don't expect different relation types to have different graph formats. So, checking only the first one should be fine.

Reply:
i don't understand the comment above the line.